python/plugins/processing/gui/ProcessingToolbox.py

```diff
@@ -374,7 +374,6 @@ def __init__(self, providerName, tree, toolbox):
 
     def refresh(self):
         self.takeChildren()
-        Processing.updateAlgsList()
```

Comment:
Wait, shouldn't this be left in place now that there's a recursion guard?

Reply:
This line is not needed. Algorithms are updated in an other way and even if there is a recursion guard this method is not the right way to update algorithm in the UI.

I have reactivated this line and tested it and algorithms are duplicated in the UI and not activated providers are displayed. So this line must remain disabled.